h5py/_hl/group.py

```diff
@@ -267,17 +263,24 @@ def create_dataset_like(self, name, other, **kwupdate):
 
         return self.create_dataset(name, **kwupdate)
 
-    def require_group(self, name):
-        # TODO: support kwargs like require_dataset
+    def require_group(self, name, track_order=None):
         """Return a group, creating it if it doesn't exist.
 
         TypeError is raised if something with that name already exists that
         isn't a group.
         """
         with phil:
             if not name in self:
-                return self.create_group(name)
+                return self.create_group(name, track_order)
             grp = self[name]
+            if track_order is None:
+                track_order = h5.get_config().track_order
+            if grp._id.get_create_plist().get_attr_creation_order() == 0:
+                order = False
+            else:
+                order = True
+            if not order == track_order:
+                raise TypeError("tack_order do not match (existing %s vs new %s)" % (order, track_order))
```

Comment:
It's not clear to me if this kind of check makes sense. `require_dataset()` checks shape & dtype - the fundamental components of what a dataset is - but all of the extra parameters (including `track_order`) are only used if it creates a new dataset.

I think it's pragmatic for details like this to be unchecked - you can tell `require_*` how to create an object if necessary, but still use an existing object even if it doesn't match all your preferences. Nobody is rushing to 'fix' the checks in `require_dataset()`, and it's nearly a year since I suggested on #897 that we leave it as-is.

Reply:
well, I will leave it as-is, just remove this kind of TODO.